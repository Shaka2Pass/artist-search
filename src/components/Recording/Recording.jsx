//Displays a lyrics of songs.
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Recording = ({ title, disambiguation, id }) => (
  <section>
    <h2><Link to={`/recording/${id}`}>{title}</Link></h2>
    <p>{disambiguation}</p>
  </section>
);

Recording.propTypes = {
  title: PropTypes.string.isRequired,
  disambiguation: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Recording;


