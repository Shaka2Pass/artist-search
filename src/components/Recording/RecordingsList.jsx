import Recording from './Recording.jsx';
import withList from '../withList';

const RecordingsList = withList(Recording, 'recordingResults');

export default RecordingsList;

