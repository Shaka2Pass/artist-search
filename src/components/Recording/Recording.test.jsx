import React from 'react';
import { shallow } from 'enzyme';
import Recording from './Recording.jsx';

describe('It describes the Recording component', () => {
  it('Matches a snapshot of Recording.jsx', () => {
    const wrapper = shallow (<Recording 
      title="Hush" 
      disambiguation="album version" />);
    expect(wrapper).toMatchSnapshot();
  });
});

