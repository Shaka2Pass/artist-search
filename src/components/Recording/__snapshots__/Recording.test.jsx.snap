// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It describes the Recording component Matches a snapshot of Recording.jsx 1`] = `
<section>
  <h2>
    <Link
      to="/recording/123123123"
    >
      Hush
    </Link>
  </h2>
  <p>
    album version
  </p>
</section>
`;
